implementations/micrometer-registry-wavefront/src/main/java/io/micrometer/wavefront/WavefrontMeterRegistry.java

```diff
@@ -86,6 +107,21 @@ private WavefrontMeterRegistry(WavefrontConfig config, Clock clock, ThreadFactor
         start(threadFactory);
     }
 
+    private WavefrontMeterRegistry(WavefrontConfig config, Clock clock, ThreadFactory threadFactory,
+                                   WavefrontSender wavefrontSender, @Nullable Tags globalTags,
+                                   Set<HistogramGranularity> histogramGranularities) {
+        super(config, clock);
+        this.config = config;
+        this.httpClient = null;
```

Comment:
Very confusing and error-prone. Both constuctors have a lot in common.
How about define super constructor for both `wavefrontSender` and `httpSender`? Something like that:
```

    public WavefrontMeterRegistry(WavefrontConfig config, Clock clock, ThreadFactory threadFactory,
	    HttpSender httpClient, @Nullable Tags globalTags) {
	this(config, clock, threadFactory, null, httpClient, null, globalTags);
	if (directToApi() && config.apiToken() == null) {
	    throw new MissingRequiredConfigurationException(
		    "apiToken must be set whenever publishing directly to the Wavefront API");
	}

	start(threadFactory);
    }

    public WavefrontMeterRegistry(WavefrontConfig config, Clock clock, ThreadFactory threadFactory,
	    WavefrontSender wavefrontSender, Set<HistogramGranularity> histogramGranularities,
	    @Nullable Tags globalTags) {
	this(config, clock, threadFactory, wavefrontSender, null, histogramGranularities, globalTags);
	start(threadFactory);
    }

    private WavefrontMeterRegistry(WavefrontConfig config, Clock clock, ThreadFactory threadFactory,
	    @Nullable WavefrontSender wavefrontSender, @Nullable HttpSender httpClient,
	    @Nullable Set<HistogramGranularity> histogramGranularities, @Nullable Tags globalTags) {
	super(config, clock);
	this.config = config;
	this.httpClient = httpClient;
	this.wavefrontSender = wavefrontSender;
	this.globalTags = globalTags;
	this.histogramGranularities = histogramGranularities;

	config().namingConvention(new WavefrontNamingConvention(config.globalPrefix()));
    }
```

Reply:
Updated. Thanks for reviewing.

Reply:
As we discussed offline, we should converge the concepts of `WavefrontSender` and `HttpSender`. That probably means a change to the wavefront SDK. Otherwise, we need to ship directly to API without the SDK until such a time as the SDK supports sender injection. Our users shouldn't be required to choose between sender customization and histogram support.